### PR TITLE
Initial promlint counter _total suffix check

### PR DESCRIPTION
**Pull request by mdlayher**
Updates #2618.  Summary check will be added in another PR.

### CODE REVIEW

File: util/promlint/promlint.go [hunk 1]
```diff
@@ -135,6 +136,30 @@ func lintMetricUnits(mf dto.MetricFamily) []Problem {
...
+	isCounter := *mf.Type == dto.MetricType_COUNTER
+	hasTotalSuffix := strings.HasSuffix(*mf.Name, "_total")
+
+	switch {
+	case isCounter && !hasTotalSuffix:
+		problems = append(problems, Problem{
+			Metric: *mf.Name,
+			Text:   `counter metrics should have "_total" suffix`,
+		})
+	case !isCounter && hasTotalSuffix:
```
Comment (brian-brazil) [inline util/promlint/promlint.go]: Might want to exempt untyped here

Comment (mdlayher) [inline util/promlint/promlint.go]: Just to clarify: it's okay for untyped metrics to have a `_total` suffix?  I have never used untyped metrics myself so I don't really know the rules around them.

If you'd like, I can exempt them from all type-specific checks, and just keep checking the help text and metric units for untyped metrics.

Comment (brian-brazil) [inline util/promlint/promlint.go]: An untyped metric might be a counter or a gauge, so we don't know which rules to apply.

The units are the same in all cases.

Comment (mdlayher) [inline util/promlint/promlint.go]: Sure thing. I'll exempt them, and also keep that in mind for any future gauge checks, if we need those.

**Comment by mdlayher** [comment]
Updated to exempt untyped metrics from suffix check.

**Comment by juliusv** [comment]
:+1:

**Pull request merged**